hnn_core/tests/test_parallel_backends.py

```diff
@@ -32,6 +33,35 @@ def _terminate_mpibackend(event, backend):
         sleep(0.01)
 
 
+def test_gid_assignment():
+    """Test that gids are assigned without overlap across ranks"""
+    hnn_core_root = op.dirname(hnn_core.__file__)
+    params_fname = op.join(hnn_core_root, 'param', 'default.json')
+    params = read_params(params_fname)
+    params.update({'N_pyr_x': 3,
+                   'N_pyr_y': 3,
+                   'tstop': 40,
+                   't_evprox_1': 5,
+                   't_evdist_1': 10,
+                   't_evprox_2': 20,
+                   'N_trials': 2})
+    net = jones_2009_model(params, add_drives_from_params=True)
+    n_drive_cells = {name: drive['n_drive_cells'] for name, drive in
+                     net.external_drives.items()}
+    n_ranks = 3
+    n_drive_cells_instantiated = dict()
+    for rank in range(n_ranks):
+        net_builder = NetworkBuilder(net)
+        net_builder._build(test_rank=rank)
```

Comment:
just test `_gid_assign`, not the whole `_build` ... and separate out the Neuron calls because you don't really have rank > 0. You are trying to create a fake rank